js/bitbay.js

```diff
@@ -140,6 +141,164 @@ module.exports = class bitbay extends Exchange {
         });
     }
 
+    async fetchMarkets (params = {}) {
+        //   { status: 'Ok',
+        //     items:
+        //     { 'BSV-USD':
+        //      { market:
+        //        { code: 'BSV-USD',
+        //          first: { currency: 'BSV', minOffer: '0.00035', scale: 8 },
+        //          second: { currency: 'USD', minOffer: '5', scale: 2 } },
+        //       time: '1557569762154',
+        //           highestBid: '52.31',
+        //       lowestAsk: '62.99',
+        //       rate: '63',
+        //       previousRate: '51.21' },
+        //      ...
+        const response = await this.v1_01PublicGetTradingTicker ({});
+        if (response['status'] !== 'Ok')
+            throw new ExchangeError (this.id + ' tickers query failed ' + this.json (response));
+        const result = [];
+        const symbols = Object.keys (response['items']);
+        for (let i = 0; i < symbols.length; i++) {
+            const symbol = symbols[i];
+            const item = response['items'][symbol];
+            const nativeMarket = item['market'];
+            const baseId = this.safeString (nativeMarket['first'], 'currency');
+            const quoteId = this.safeString (nativeMarket['second'], 'currency');
+            const id = baseId + quoteId;
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const precision = this.safeInteger (nativeMarket['second'], 'scale');
+            // todo: check that the limits have ben interpreted correctly
+            // todo: parse the fees page
+            result.push ({
+                'id': id,
+                'symbol': base + '/' + quote,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'precision': precision,
+                'active': undefined,
+                'fee': undefined,
+                'limits': {
+                    'amount': {
+                        'min': this.safeFloat (nativeMarket['first'], 'minOffer'),
+                    },
+                    'cost': {
+                        'min': this.safeFloat (nativeMarket['second'], 'minOffer'),
+                    },
+                },
+                'info': item,
+            });
+        }
+        return result;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = this.extend ({
+            'markets': symbol ? [this.marketId (symbol)] : [],
+        }, params);
+        const response = await this.v1_01PrivateGetTradingHistoryTransactions ({ 'query': this.json (request) });
+        if (response['status'] !== 'Ok')
```

Comment:
This has to be handled in `handleErrors` – need to move these checks there, or make sure everything is handled there properly and remove the conditionals for `response['status']` from all `fetch*` methods.